### PR TITLE
[New Rule] Potential Process Injection via PowerShell

### PR DESCRIPTION
**Pull request by w0rk3r**
## Summary

Detects the use of Windows API Functions and Assembly.Load Method that Malware and Hacktools commonly abuse to
Inject into process and load binaries and DLLs into memory in Powershell Scripts. Attackers use these functions to evade
detections by loading executables and DLLs in memory and injecting them into legit processes.

## Contributor checklist

- Have you signed the [contributor license agreement](https://www.elastic.co/contributor-agreement)?
- Have you followed the [contributor guidelines](https://github.com/elastic/detection-rules/blob/main/CONTRIBUTING.md)?

### CODE REVIEW

File: rules/windows/defense_evasion_posh_process_injection.toml [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+    LoadLibraryEx or
+    GetProcAddress or
+    OpenProcess or
+    VirtualProtect) and
+   (CreateRemoteThread or
+    NtCreateThreadEx or
+    CreateThread or
+    QueueUserAPC or
+    (SuspendThread and ResumeThread))
+  )
```
Comment (Samirbous) [inline rules/windows/defense_evasion_posh_process_injection.toml:50]:
```suggestion
event.code:"4104" and 
  powershell.file.script_block_text : (
   (VirtualAlloc or
    VirtualAllocEx or 
    VirtualProtect or
    LdrLoadDll or
    LoadLibrary or
    LoadLibraryA or
    LoadLibraryEx or
    GetProcAddress or
    OpenProcess or 
    OpenProcessToken or
    AdjustTokenPrivileges) and
   (WriteProcessMemory or
    CreateRemoteThread or
    NtCreateThreadEx or
    CreateThread or
    QueueUserAPC or
    SuspendThread or 
    ResumeThread)
  )
```

File: rules/windows/defense_evasion_posh_process_injection.toml [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+creation_date = "2021/10/14"
+maturity = "production"
+updated_date = "2021/10/14"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule detects the use of Windows API Functions that Malware and Hacktools commonly abuse to Inject into process and
+load binaries and DLLs into memory in Powershell Scripts. Attackers use these functions to evade detections by loading
+executables and DLLs in memory and injecting them into legit processes.,
```
Comment (Samirbous) [inline rules/windows/defense_evasion_posh_process_injection.toml:11]:
```suggestion
This rule detects the use of Windows API functions that are commonly abused by Malware and Hacktools to load 
malicious code or injecting it into remote processes.
```

File: rules/windows/defense_evasion_posh_process_injection.toml [hunk 3]
```diff
@@ -0,0 +1,77 @@
+[metadata]
+creation_date = "2021/10/14"
+maturity = "production"
+updated_date = "2021/10/14"
+
+[rule]
+author = ["Elastic"]
+description = """
+This rule detects the use of Windows API functions that are commonly abused by Malware and Hacktools to load 
+malicious code or injecting it into remote processes.
```
Comment (brokensound77) [inline rules/windows/defense_evasion_posh_process_injection.toml:10]:
```suggestion
Detects the use of Windows API functions that are commonly abused by malware and security tools to load 
malicious code or inject it into remote processes.
```

File: rules/windows/defense_evasion_posh_process_injection.toml [hunk 4]
```diff
@@ -0,0 +1,77 @@
...
+    OpenProcessToken or
+    AdjustTokenPrivileges) and
+   (WriteProcessMemory or
+    CreateRemoteThread or
+    NtCreateThreadEx or
+    CreateThread or
+    QueueUserAPC or
+    SuspendThread or 
+    ResumeThread)
+  )
```
Comment (brokensound77) [inline rules/windows/defense_evasion_posh_process_injection.toml:51]: formatting nit:

```suggestion
event.code:"4104" and 
  powershell.file.script_block_text : (
   (VirtualAlloc or VirtualAllocEx or VirtualProtect or LdrLoadDll or LoadLibrary or LoadLibraryA or
      LoadLibraryEx or GetProcAddress or OpenProcess or OpenProcessToken or AdjustTokenPrivileges) and
   (WriteProcessMemory or CreateRemoteThread or NtCreateThreadEx or CreateThread or QueueUserAPC or
      SuspendThread or ResumeThread)
  )
```

**Review by brokensound77 — APPROVED** [review]
Few changes then LGTM - nice job 👍

**Pull request merged**